### ISSUE TITLE
Cannot update form after initial creation

### ISSUE THREAD

**Issue by tcrawford-figure** [auth0/terraform-provider-auth0#1054]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have upgraded to the [latest version](https://github.com/auth0/terraform-provider-auth0/releases/latest) of this provider and the issue still persists.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

When modifying an existing `auth0_form` configuration and running apply I am receiving the following error, despite making no changes to any node ids:

>Error: 400 Bad Request: Some nodes point to unknown node IDs

I also want to take time to point out that documentation in this area feels lackluster. I had to piece together what I can export from the Auth0 Forms UI and pull requests to the Auth0 API from the browser for form and flow updates in order to guess at what the correct configuration would even be in the first place for forms. After getting something successful after about 10 or so tries, I found a need to modify the form that I got in and find myself here.

### Expectation

When I modify an `auth0_form` resource without modifying the nodes I expect it to update in place. Even if the node values are updated, I expect the form to override the existing node chain.

### Reproduction

Given:
I have updated a form configuration, like the coordinates for example.

When:
I apply my plan

Then:
My changes should be applied successfully.

### Auth0 Terraform Provider version

1.7.1

### Terraform version

1.7.5

**Comment by aescudero-abbott** [comment]
I'm also facing the same issue

**Comment by msinstil** [comment]
I'm also facing the same issue, this makes the forms unusable from terraform. Please fix!!

**Comment by msinstil** [comment]
> Thanks for opening the issue. Please use Terraform version `1.9.x` for better support of `jsonencode`. If the issue persists, please share a sample `auth0_form` to investigate more.

Thanks @kushalshit27 for looking into this. 

I am using Terraform v1.9.5 and Auth0 provider v1.7.1

Issue can be reproduced by taking the example form from Auth0 Provider docs (below), deploying it and then trying to add any new components to the nodes like another FIELD, BLOCK, HTML, RICH_TEXT etc.

```
# Example:
resource "auth0_form" "my_form" {
  name = "My KYC Form"

  start = jsonencode({
    coordinates = {
      x = 0
      y = 0
    }
    next_node = "step_ggeX"
  })
  nodes = jsonencode([{
    alias = "New step"
    config = {
      components = [{
        category = "FIELD"
        config = {
          max_length = 50
          min_length = 1
          multiline  = false
        }
        id        = "full_name"
        label     = "Your Name"
        required  = true
        sensitive = false
        type      = "TEXT"
        }, {
        category = "BLOCK"
        config = {
          text = "Continue"
        }
        id   = "next_button_3FbA"
        type = "NEXT_BUTTON"
      }]
      next_node = "$ending"
    }
    coordinates = {
      x = 500
      y = 0
    }
    id   = "step_ggeX"
    type = "STEP"
  }])
  ending = jsonencode({
    after_submit = {
      flow_id = "<my_flow_id>" # Altenative ways: (flow_id = auth0_flow.my_flow.id) or using terraform variables
    }
    coordinates = {
      x = 1250
      y = 0
    }
    resume_flow = true
  })

  style = jsonencode({
    css = "h1 {\n  color: white;\n  text-align: center;\n}"
  })

  translations = jsonencode({
    es = {
      components = {
        rich_text_uctu = {
          config = {
            content = "<h2>Help us verify your personal information</h2><p>We want to learn more about you so that we can validate and protect your account...</p>"
          }
        }
      }
      messages = {
        custom = {}
        errors = {
          ERR_ACCEPTANCE_REQUIRED = "Por favor, marca este campo para continuar."
        }
      }
    }
  })

  languages {
    default = "en"
    primary = "en"
  }
}
```

**Comment by GDownes** [comment]
I'm facing the same issue

**Comment by tcrawford-figure** [comment]
@duedares-rvj I noticed that `1.7.2` was attempted to be released but it looks like the [release workflow](https://github.com/auth0/terraform-provider-auth0/actions/runs/11608001065) had an issue during release that kept it from publishing to the [terraform registry](https://registry.terraform.io/providers/auth0/auth0/latest/docs). I just wanted to make sure someone had visibility to this. Thank you!

**Comment by duedares-rvj** [comment]
@tcrawford-figure That's right. We are making some required updated to goreleaser for this, shall revert back on the publish is successful.

**Comment by duedares-rvj** [comment]
@tcrawford-figure We release https://github.com/auth0/terraform-provider-auth0/releases/tag/v1.7.3 with the required changes. :) ^_^

**Comment by tcrawford-figure** [comment]
@duedares-rvj Perfect! Thank you so much!
